pulsar-common/src/test/java/org/apache/pulsar/common/util/keystoretls/JettySslContextFactoryWithAutoRefreshTest.java

```diff
@@ -0,0 +1,186 @@
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@Slf4j
+public class JettySslContextFactoryWithAutoRefreshTest {
+
+    @Test
+    public void testJettyTlsServerTls() throws Exception {
+        Configurator.setRootLevel(Level.INFO);
+        Server server = new Server();
+        List<ServerConnector> connectors = new ArrayList<>();
+        SslContextFactory.Server factory = KeyStoreSSLContext.createSslContextFactory(null,
+                "JKS", Resources.getResource("ssl/jetty_server_key.jks").getPath(),
+                "jetty_server_pwd", false, "JKS",
+                Resources.getResource("ssl/jetty_server_trust.jks").getPath(),
+                "jetty_server_pwd", true, null,
+                null, 600);
+        factory.setHostnameVerifier((s, sslSession) -> true);
+        ServerConnector connector = new ServerConnector(server, factory);
+        connector.setPort(0);
+        connectors.add(connector);
+        server.setConnectors(connectors.toArray(new ServerConnector[0]));
+        server.start();
+        // client connect
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+        registryBuilder.register("https", new SSLConnectionSocketFactory(getClientSslContext(), new NoopHostnameVerifier()));
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(registryBuilder.build());
+        httpClientBuilder.setConnectionManager(cm);
+        CloseableHttpClient httpClient = httpClientBuilder.build();
+        HttpGet httpGet = new HttpGet("https://localhost:" + connector.getLocalPort());
+        httpClient.execute(httpGet);
+    }
+
+    @Test(expectedExceptions = SSLHandshakeException.class)
+    public void testJettyTlsServerInvalidTlsProtocol() throws Exception {
+        Configurator.setRootLevel(Level.INFO);
+        Server server = new Server();
+        List<ServerConnector> connectors = new ArrayList<>();
+        SslContextFactory.Server factory = KeyStoreSSLContext.createSslContextFactory(null,
+                "JKS", Resources.getResource("ssl/jetty_server_key.jks").getPath(),
+                "jetty_server_pwd", false, "JKS",
+                Resources.getResource("ssl/jetty_server_trust.jks").getPath(),
+                "jetty_server_pwd", true, null,
+                new HashSet<String>() {
+                    {
+                        this.add("TLSv1.3");
+                    }
+                }, 600);
+        factory.setHostnameVerifier((s, sslSession) -> true);
+        ServerConnector connector = new ServerConnector(server, factory);
+        connector.setPort(0);
+        connectors.add(connector);
+        server.setConnectors(connectors.toArray(new ServerConnector[0]));
+        server.start();
```

Comment:
Please stop all the servers and close all the clients

Reply:
fixed